src/main/java/frc/robot/vision/filters/Filter.java

```diff
@@ -0,0 +1,45 @@
+package frc.robot.vision.filters;
+
+import frc.robot.vision.rawdata.RawVisionData;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.Arrays;
+import java.util.function.Function;
+
+public class Filter<T extends RawVisionData> {
+
+	private final Function<T, Boolean> filteringFunction;
+
+	public Filter(Function<T, Boolean> filteringFunction) {
+		this.filteringFunction = filteringFunction;
+	}
+
+	public boolean applyFilter(T data) {
+		return filteringFunction.apply(data);
+	}
+
+	public Filter<T> and(Filter<T> secondFilter) {
+		return new Filter<>(data -> secondFilter.applyFilter(data) && applyFilter(data));
+	}
+
+	public Filter<T> or(Filter<T> anotherFilter) {
+		return new Filter<>(data -> anotherFilter.applyFilter(data) || applyFilter(data));
+	}
+
+	public void logFilter(String logPath, T data) {
+		Logger.recordOutput(logPath, applyFilter(data));
+	}
+
+	@SafeVarargs
+	public static <T extends RawVisionData> Filter<T> combineFilters(Filter<T>... filters) {
+		return new Filter<>(data -> Arrays.stream(filters).allMatch(filter -> filter.applyFilter(data)));
+	}
+
+	@SafeVarargs
+	public static <T extends RawVisionData> void logFilters(String logPath, T data, Filter<T>... filters) {
```

Comment:
I'd like to have a version of this function that takes pairs of filters and logpaths. Having just a number to indicate if certain filter is passing or not is not that useful for debugging.

Reply:
the logging should be done when you apply a function and not inside of a class. thats why we dont have a log func in every single one